src/ripple/rpc/handlers/GetCounts.cpp

```diff
@@ -58,58 +58,50 @@ textTime(std::string& text, UptimeClock::time_point& seconds,
         text += "s";
 }
 
-// {
-//   min_count: <number>  // optional, defaults to 10
-// }
-Json::Value doGetCounts (RPC::Context& context)
+Json::Value getCountsJson(Application& app, int minObjectCount)
 {
-    int minCount = 10;
+    auto objectCounts = CountedObjects::getInstance().getCounts(minObjectCount);
 
-    if (context.params.isMember (jss::min_count))
-        minCount = context.params[jss::min_count].asUInt ();
-
-    auto objectCounts = CountedObjects::getInstance ().getCounts (minCount);
-
-    Json::Value ret (Json::objectValue);
+    Json::Value ret(Json::objectValue);
 
     for (auto const& it : objectCounts)
     {
         ret [it.first] = it.second;
     }
 
-    int dbKB = getKBUsedAll (context.app.getLedgerDB ().getSession ());
+    int dbKB = getKBUsedAll (app.getLedgerDB ().getSession ());
 
     if (dbKB > 0)
         ret[jss::dbKBTotal] = dbKB;
 
-    dbKB = getKBUsedDB (context.app.getLedgerDB ().getSession ());
+    dbKB = getKBUsedDB (app.getLedgerDB ().getSession ());
 
     if (dbKB > 0)
         ret[jss::dbKBLedger] = dbKB;
 
-    dbKB = getKBUsedDB (context.app.getTxnDB ().getSession ());
+    dbKB = getKBUsedDB (app.getTxnDB ().getSession ());
 
     if (dbKB > 0)
         ret[jss::dbKBTransaction] = dbKB;
 
     {
-        std::size_t c = context.app.getOPs().getLocalTxCount ();
+        std::size_t c = app.getOPs().getLocalTxCount ();
         if (c > 0)
             ret[jss::local_txs] = static_cast<Json::UInt> (c);
     }
 
-    ret[jss::write_load] = context.app.getNodeStore ().getWriteLoad ();
+    ret[jss::write_load] = app.getNodeStore ().getWriteLoad ();
 
     ret[jss::historical_perminute] = static_cast<int>(
-        context.app.getInboundLedgers().fetchRate());
-    ret[jss::SLE_hit_rate] = context.app.cachedSLEs().rate();
-    ret[jss::node_hit_rate] = context.app.getNodeStore ().getCacheHitRate ();
-    ret[jss::ledger_hit_rate] = context.app.getLedgerMaster ().getCacheHitRate ();
-    ret[jss::AL_hit_rate] = context.app.getAcceptedLedgerCache ().getHitRate ();
+        app.getInboundLedgers().fetchRate());
+    ret[jss::SLE_hit_rate] = app.cachedSLEs().rate();
+    ret[jss::node_hit_rate] = app.getNodeStore ().getCacheHitRate ();
+    ret[jss::ledger_hit_rate] = app.getLedgerMaster ().getCacheHitRate ();
+    ret[jss::AL_hit_rate] = app.getAcceptedLedgerCache ().getHitRate ();
 
-    ret[jss::fullbelow_size] = static_cast<int>(context.app.family().fullbelow().size());
-    ret[jss::treenode_cache_size] = context.app.family().treecache().getCacheSize();
-    ret[jss::treenode_track_size] = context.app.family().treecache().getTrackSize();
+    ret[jss::fullbelow_size] = static_cast<int>(app.family().fullbelow().size());
```

Comment:
I don't understand why we are casting to `int` instead of `Json::UInt`. I know the rest of the function does something similar, so I get why you did it. I'm not understanding why it was done origionally.

Reply:
I didn't actually touch any logic in this code, just pulled the core of it from `doGetCounts` to `getCountsJson` so that it could be called from `OverlayImpl`. I tried to keep the formatting identical so the diff shows that just `context.` was the part that was removed.